### DISCUSSION TITLE
Split bars to allow split drag and drop, closing per mouse, and more

### DISCUSSION THREAD

**Discussion by SenseiMarv** [ghostty-org/ghostty#3747] [Feature Requests, Ideas]
Coming from iTerm2, I'm very used to having bars not only for tabs but also for splits:

![image](https://github.com/user-attachments/assets/b5953367-ab7c-4dc7-a85a-31869a809b7e)

The bar allows me to quickly close a split with the mouse, but also to drag and drop it around, double click to quickly full screen it and access more options of the split:

Drag and Drop:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/7f74470e-605e-43e5-a956-21b9421cc2ff" />

Quick options:
<img width="1108" alt="image" src="https://github.com/user-attachments/assets/6374a808-5e4a-4e18-ab0b-ce44de9be11e" />

As an iTerm2 user, this is probably the biggest thing I'm missing in Ghostty at the moment, so I'm interested in gathering feedback to adapt something similar.

**Comment by SenseiMarv** [comment]
PS: This would also be a possible solution for questions already asked about how to move splits, e.g.: https://github.com/ghostty-org/ghostty/discussions/3325

**Comment by pluiedev** [comment]
Konsole also has this. Definitely viable as an option.

**Reply by rhodes-b** [reply]
My understanding was jcollie said this would be difficult to do in GTK

**Reply by pluiedev** [reply]
It would be difficult, but not impossible

**Comment by bangbangsheshotmedown** [comment]
+1, i can do this with Terminator, would love to switch but not being able to drag&drop splits is deal breaker


https://github.com/user-attachments/assets/ae09f5d1-b302-4396-88a9-31acefccaf0a

**Comment by blurayne** [comment]
You can also do in tilix and it's the only thing that prevents me from using GhostTTY.

<img width="944" height="922" alt="image" src="https://github.com/user-attachments/assets/b95c3f2c-7d9e-4cd3-91e9-eb976a2641c6" />

You can also _not only_  drag and drop BUT pull out a "split" into a window and back again. Nyce and useful sometimes when presenting something ;)

I used terminator a before I switched to tilix some years ago. Really didn't like tiling window managers or stuff like alacritty. Not even Wezterm could keep from using Tilix again.
